### PR TITLE
Start of OCS Share API sharees endpoint

### PR DESCRIPTION
**Pull request by rullzer**
Start of implementation of #18117

TODO:
- [ ] Tests, tests, tests and tests
- [ ] Fix pagination (OCS does merging which breaks this maybe later)
- [x] Sort results before splicing
- [ ] Add filtering based on provided share id's (or provided list of users/groups etc??)
- [ ] Did I mention tests?

---
### Todo
- [x] Make share dropdown call the OCS endpoint
- [ ] Add tests for `search()` directly
- [x] Add a filter that prevents sharing with the same user/group/remote again

@DeepDiver1975 FYI

### CODE REVIEW

**Comment by karlitschek** [comment]
@rullzer Thanks a lot. Looks good so far :-)

File: apps/files_sharing/tests/sharees.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+use OCA\Files_Sharing\API\Sharees;
```
Comment (nickvergessen) [inline apps/files_sharing/tests/sharees.php]: missing namespace and copyright

File: apps/files_sharing/api/sharees.php [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	 *
+	 * @return array possible sharees
+	 */
+	private function getUsers($search, $shareWithGroupOnly) {
+		$sharees = [];
+		
+		$users = [];
+		if ($shareWithGroupOnly) {
+			// Search in all the groups this user is part of
+			$userGroups = $this->groupManager->getUserGroupsIds($this->userSession->getUser());
```
Comment (nickvergessen) [inline apps/files_sharing/api/sharees.php]: PHP Fatal error:  Call to undefined method Mock_IGroupManager_6a976b4f::getUserGroupsIds() in /home/nickv/ownCloud/master/core/apps/files_sharing/api/sharees.php on line 82

Comment (rullzer) [inline apps/files_sharing/api/sharees.php]: Ah crap typo... ```getUserGroupsIds``` -> ```getUserGroupIds```

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by nickvergessen** [comment]
@owncloud-bot retest this please

**Comment by nickvergessen** [comment]
Replaced by #18234

**Pull request closed without merging**